plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/exporter/AbstractExporter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.core.exporter;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.plugin.insights.rules.model.SearchQueryRecord;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Base Abstract class for Query Insights exporters
+ */
+public abstract class AbstractExporter {
+    /**
+     * Logger of exporter
+     */
+    protected final Logger logger = LogManager.getLogger(this.getClass());
+
+    /**
+     * Constructor of AbstractExporter
+     */
+    protected AbstractExporter() {}
```

Comment:
Since we enabled JavaDoc check for query insights plugin, this is needed to make the checker happy. Otherwise we get
```
AbstractExporter.java:21: error: AbstractExporter.<init> (constructor): javadocs are missing
public abstract class AbstractExporter {
```